src/avro_copy.cpp

```diff
@@ -403,9 +440,12 @@ WriteAvroGlobalState::WriteAvroGlobalState(ClientContext &context, FunctionData
 		json_metadata = bind_data.json_metadata.c_str();
 	}
 
-	while ((ret = avro_file_writer_create_from_writers_with_metadata(writer, datum_writer, bind_data.schema, &file_writer, json_metadata)) == ENOSPC) {
+	while ((ret = avro_file_writer_create_from_writers_with_metadata(writer, datum_writer, bind_data.schema,
+	                                                                 &file_writer, json_metadata)) == ENOSPC) {
 		auto current_capacity = memory_buffer.GetCapacity();
 		memory_buffer.Resize(NextPowerOfTwo(current_capacity * 2));
+		// re-initialize writer to use correct data location
+		writer = avro_writer_memory(const_char_ptr_cast(memory_buffer.GetData()), memory_buffer.GetCapacity());
```

Comment:
Good catch, but I don't think we can re-initialize the writer
This is what we do elsewhere:
```c++
	while ((ret = avro_file_writer_flush(global_state.file_writer)) == ENOSPC) {
		auto current_capacity = buffer.GetCapacity();
		buffer.Resize(NextPowerOfTwo(current_capacity * 2));
		avro_writer_memory_set_dest(global_state.writer, (const char *)buffer.GetData(), buffer.GetCapacity());
	}
```

I think we also need to use `avro_writer_memory_set_dest` here

Reply:
Actually.. this is when creating the `file_writer` fails right?
So it hasn't taken ownership of the `writer`, so it has to be destroyed first, then reinitialized?